src/creature_states_spdig.c

```diff
@@ -320,7 +320,8 @@ long check_out_unclaimed_spells(struct Thing *spdigtng, long range)
           break;
         i = thing->next_of_class;
         // Per-thing code
-        if (thing_is_spellbook(thing) || thing_is_special_box(thing))
+        if ((thing_is_spellbook(thing) || thing_is_special_box(thing)) && 
+            thing_can_be_picked_to_place_in_player_room_of_role(thing, spdigtng->owner, RoRoF_PowersStorage, TngFRPickF_Default))
```

Comment:
I suppose the (thing_is_spellbook(thing) || thing_is_special_box(thing)) is unneeded as thing_can_be_picked_to_place_in_player_room_of_role should cover it

Reply:
> What about crates? If mapmaker gives the flag to crate, it will still be able to be dragged on right? Altho I can't imagine a scenario where mapmaker would want a not draggable crate. 🤔

check_out_unclaimed_traps already uses thing_can_be_picked_to_place_in_player_room_of_role so should be ok

Reply:
> Another thing, what about gold pot or gold pile? Not technically draggable but the name `IGNORED_BY_IMPS` makes me think it could work if I were to set this flag on those objects, but it doesn't of course, the flag name is confusing.

looks like get_next_unclaimed_gold_thing_pickable_by_digger has it as a TODO but indeed it could also make use of thing_can_be_picked_to_place_in_player_room_of_role

Reply:
as far as tested, this works with crates, spellbooks, and special boxes, also in possession mode

Reply:
> looks like get_next_unclaimed_gold_thing_pickable_by_digger has it as a TODO but indeed it could also make use of thing_can_be_picked_to_place_in_player_room_of_role

yes, I think mapmakers could appreciate having this option